### ISSUE TITLE
False-positive in Indentation check with yield switch statements

### ISSUE THREAD

**Issue by archer-321** [checkstyle/checkstyle#19709]
I have read the check documentation: https://checkstyle.org/checks/misc/indentation.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the CLI and pasted its output below, as the output describes the problem better than 1,000 words.

```bash
$ javac Example.java
```

`$ cat example-config.xml`
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="Indentation"/>
    </module>
</module>
```

`$ cat Example.java`
```java
class Example {
    void foo() {
        int i = 0;
        int j = switch (i) {
            case 0 -> {
                yield switch (i) {
                    case 0 -> 0;  // error: expected level should be 24
                    default -> 1; // error: expected level should be 24
                };                // error: expected level should be 20
            }
            default -> 1;
        };
    }
}
```

``` bash
$ RUN_LOCALE="-Duser.language=en -Duser.country=US"
$ java $RUN_LOCALE -jar checkstyle-13.4.0-all.jar -c example-config.xml Example.java
Starting audit...
[ERROR] /tmp/Example.java:7:21: 'case' child has incorrect indentation level 20, expected level should be 24. [Indentation]
[ERROR] /tmp/Example.java:8:21: 'case' child has incorrect indentation level 20, expected level should be 24. [Indentation]
[ERROR] /tmp/Example.java:9:17: 'switch rcurly' has incorrect indentation level 16, expected level should be 20. [Indentation]
Audit done.
Checkstyle ends with 3 errors.
```

---

Expected behaviour:\
The snippet
```java
yield switch (i) {
    case 0 -> 0;
    default -> 1;
};
```
should already be formatted correctly. Instead, Checkstyle expects the snippet to be formatted as
```java
yield switch (i) {
        case 0 -> 0;
        default -> 1;
    };
```

---

I tried checking for duplicate issues; in particular, this seems similar to checkstyle/checkstyle#17167 and checkstyle/checkstyle#15967.
However, considering both issues are closed, I went ahead and opened a new one.

Thank you for maintaining this project!

**Comment by Brijeshthummar02** [comment]
on it

**Comment by archer-321** [comment]
I've just updated to Checkstyle 13.4.2 and can confirm the false-positive is fixed in my codebase. Thank you, @Brijeshthummar02, for implementing this fix ❤️

**Comment by Brijeshthummar02** [comment]
> I've just updated to Checkstyle 13.4.2 and can confirm the false-positive is fixed in my codebase. Thank you, [@Brijeshthummar02](https://github.com/Brijeshthummar02), for implementing this fix ❤️

Awesome!!
Pleasure to work around, feel free to ping me in future for such issues.

**Comment by romani** [comment]
@archer-321, if you project is open source please let us know, we can use it at regression testing.
It is hard to find projects that use modern java and Indentation.

**Comment by archer-321** [comment]
> if you project is open source please let us know, we can use it at regression testing.

The project I encountered the false-positive in is owned by someone else, so it's unfortunately not up to me to make it open source. I will, however, keep regression testing in mind when configuring Checkstyle in open source projects in the future 👍
